afs/models.py

```diff
@@ -196,39 +209,95 @@ def upload_model(
         # Fetch tags apm_node
         pai_data_dir = os.getenv("PAI_DATA_DIR", None)
         if pai_data_dir:
-            pai_data_dir = json.loads(pai_data_dir)
-            if "data" in pai_data_dir:
-                data = pai_data_dir["data"]
-                if "machineIdList" in data:
-                    machineIdList = data.get('machineIdList', [None]).pop(0)
-                    if machineIdList:
-                        tags.update({"apm_node": str(machineIdList)})
+            try:
+                pai_data_dir = json.loads(pai_data_dir)
+                if "data" in pai_data_dir and "type" in pai_data_dir:
+                    data = pai_data_dir["data"]
+                    firehose_type = pai_data_dir["type"]
+
+                    if "machineIdList" in data and firehose_type == "apm-firehose":
+                        machineIdList = data.get("machineIdList", [None]).pop(0)
+
+                        if machineIdList:
+                            tags.update({"apm_node": str(machineIdList)})
+            except Exception as e:
+                print(
+                    f"PAI_DATA_DIR value is not valid json format for apm_node. Exception:{e}, Value: {pai_data_dir}"
+                )
 
         # evaluation result
         evaluation_result.update(extra_evaluation)
         data = dict(
             tags=json.dumps(tags), evaluation_result=json.dumps(evaluation_result)
         )
-        files = {"model": model_file}
-
+        # model name
         if model_name:
             self._naming_rule(model_name)
             data.update({"name": model_name})
 
         extra_paths = [self.repo_id, self.sub_entity_uri]
-        resp = self._create(
-            data=data, files=files, extra_paths=extra_paths, form="data"
-        )
+        # Load model size < 300 MB
+        file_size = os.path.getsize(model_path)
+        if file_size < (300 * 1024 * 1024) and not blob_mode:
+            with open(model_path, "rb") as f:
+                model_file = BytesIO(f.read())
+            model_file.seek(0)
+            files = {"model": model_file}
+
+            resp = self._create(
+                data=data, files=files, extra_paths=extra_paths, form="data"
+            )
+
+        # between 300M - 1G model file
+        elif file_size < (1024 * 1024 * 1024) or blob_mode:
+            if not (
+                self._blob_endpoint
+                and self._blob_accessKey
+                and self._blob_secretKey
+                and self._bucket_name
+            ):
+                raise ValueError(
+                    f"Blob information is not enough to put object to blob, {self._blob_endpoint}, {self._blob_accessKey}, {self._blob_secretKey}, {self._bucket_name}"
```

Comment:
Do we drop support of Python-3.5?

Reply:
These are my miss to write f-string.
However, AFS notebook image support Python-3.6.8.
My boss said that afs2-model can support 3.6, 3.7.
I will fix the readme.